extensions/cornerstone-3d/src/services/SyncGroupService/SyncGroupService.ts

```diff
@@ -0,0 +1,101 @@
+import { synchronizers, SynchronizerManager } from '@cornerstonejs/tools';
+
+import { pubSubServiceInterface } from '@ohif/core';
+
+const EVENTS = {
+  TOOL_GROUP_CREATED:
+    'event::cornerstone-3d::syncgroupservice:toolgroupcreated',
+};
+
+export type SyncGroup = {
+  type: string;
+  id: string;
+  source: boolean;
+  target: boolean;
+};
+
+const POSITION = 'cameraposition';
+const VOI = 'voi';
+
+export default class SyncGroupService {
+  serviceManager: any;
+  listeners: { [key: string]: (...args: any[]) => void } = {};
+  EVENTS: { [key: string]: string };
+
+  constructor(serviceManager) {
+    this.serviceManager = serviceManager;
+    this.listeners = {};
+    this.EVENTS = EVENTS;
+    //
+    Object.assign(this, pubSubServiceInterface);
+  }
+
+  private _createSynchronizer(type: string, id: string) {
+    type = type.toLowerCase();
+    if (type === POSITION) {
+      return synchronizers.createCameraPositionSynchronizer(id);
+    } else if (type === VOI) {
+      return synchronizers.createVOISynchronizer(id);
+    }
+  }
+
+  public addViewportToSyncGroup(
+    viewportId: string,
+    renderingEngineId: string,
+    syncGroups?: SyncGroup[]
+  ): void {
+    if (!syncGroups || !syncGroups.length) {
+      return;
+    }
+
+    syncGroups.forEach(syncGroup => {
+      const { type, id, target, source } = syncGroup;
+
+      let synchronizer = SynchronizerManager.getSynchronizer(id);
+
+      if (!synchronizer) {
+        synchronizer = this._createSynchronizer(type, id);
```

Comment:
We are creating synchronizers as we see new `id`s, but we are never deleting them. Should we also lazily delete them when we remove the last viewport from a syncgroup in `removeViewportFromSyncGroup`?